File: src/main/java/com/sstep/recipe/manager/entity/Recipe.java
package com.sstep.recipe.manager.entity;

import com.sstep.recipe.manager.dto.RecipeData;
import com.sstep.recipe.manager.enums.RecipeType;

import javax.persistence.*;


@Entity
@Table(name = "recipe")
public class Recipe {

    private Long id;
    private String name;
    private String description;
    private RecipeType recipeType;
    private RecipeData recipeData;

    public Recipe() {
    }

    @Basic
    @Column(name = "id")
    @Id
    @SequenceGenerator(name = "recipe_seq_gen", sequenceName = "recipe_seq", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "recipe_seq_gen")
    public Long getId() {
        return id;
    }

    public void setId(final Long id) {
        this.id = id;
    }

    @Basic
    @Column(name = "name", nullable = false)
    public String getName() {
        return name;
    }

    public void setName(final String name) {
        this.name = name;
    }

    @Basic
    @Column(name = "description")
    public String getDescription() {
        return description;
    }

    public void setDescription(final String description) {
        this.description = description;
    }

    @Basic
    @Column(name = "type", nullable = false)
    public RecipeType getRecipeType() {
        return recipeType;
    }

    public void setRecipeType(final RecipeType recipeType) {
        this.recipeType = recipeType;
    }

    @Column(name = "data")
    public RecipeData getRecipeData() {
        return recipeData;
    }

    public void setRecipeData(final RecipeData recipeData) {
        this.recipeData = recipeData;
    }
}

File: src/main/kotlin/com/sstep/recipe/manager/entity/Recipe.kt
package com.sstep.recipe.manager.entity

import com.sstep.recipe.manager.dto.RecipeData
import com.sstep.recipe.manager.enums.RecipeType
import jakarta.persistence.*

@Entity
@Table(name = "recipe")
class Recipe {

    @get:Id
    @get:Column(name = "id")
    @get:Basic
    @get:GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "recipe_seq_gen")
    @get:SequenceGenerator(name = "recipe_seq_gen", sequenceName = "recipe_seq", allocationSize = 1)
    var id: Long? = null

    @get:Basic
    @get:Column(name = "name", nullable = false)
    var name: String? = null

    @get:Basic
    @get:Column(name = "description")
    var description: String? = null

    @get:Basic
    @get:Column(name = "type", nullable = false)
    var recipeType: RecipeType? = null

    @get:Column(name = "data")
    var recipeData: RecipeData? = null

    @get:Basic
    @get:Column(name = "preview_id")
    var previewId: String? = null
}